pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerService.java

```diff
@@ -171,6 +171,9 @@ public void start(URI dlogUri,
             this.functionMetaDataManager = new FunctionMetaDataManager(
                     this.workerConfig, this.schedulerManager, this.client, errorNotifier);
 
+            // initialize function metadata manager
+            this.functionMetaDataManager.initialize();
```

Comment:
Why do you need to move this here?

Reply:
Upon start, we always first want to get up-to-date before becoming leader and starting to serve requests

Reply:
This should happen after membership manager is created/started.  We want to do this prior to the worker "joining" the cluster

Reply:
Whats the harm before doing that? All that will happen is that this will read the topic and update its internal state